components/script/dom/bindings/root.rs

```diff
@@ -521,8 +522,7 @@ impl<T> Clone for Dom<T> {
 impl<T> Clone for LayoutDom<'_, T> {
     #[inline]
     fn clone(&self) -> Self {
-        assert_in_layout();
-        LayoutDom { value: self.value }
+        *self
```

Comment:
I think this change was talked about in https://github.com/servo/servo/pull/31894#discussion_r1541232929 but it wasn't applied in the end. In that discussion, @mrobinson proposed doing this instead to avoid removing the assert:

```suggestion
        assert_in_layout();
        { *self }
```

Reply:
Thanks, @eerii, very helpful! 

I double checked for this particular warning, it still persisted after proposed change. 

Since [assert_in_layout() needed to be kept](https://github.com/servo/servo/pull/31894#discussion_r1541232929), I decided to allow it. 

Considering those changes and formatting, I decided to create a new commit rather that committing your suggestion, if you don't mind :)

Reply:
You're right, sorry for not realizing before! The error still persist because `Clone` has the assert but `Copy` doesn't, see https://github.com/servo/servo/pull/31894#discussion_r1541254417. Maybe the `Copy` trait should be removed like it says in the comment? Probably a good idea to wait to hear from a maintainer to see what to do, since this is a bit trickier than it seemed.

And no problem at all about the commit!

Reply:
It does sound like the appropriate thing to do here might be to remove the Copy trait implementation...but only if that's possible. Does Servo still compile with that removed?

Reply:
@mrobinson removing Copy trait implementation (`impl<T> Copy for LayoutDom<'_, T> {}`) caused following compile errors:
```
error[E0204]: the trait `std::marker::Copy` cannot be implemented for this type
  --> components/script/layout_dom/document.rs:34:54
   |
23 |     document: LayoutDom<'dom, Document>,
   |     ----------------------------------- this field does not implement `std::marker::Copy`
...
34 | impl<'dom, LayoutDataType: LayoutDataTrait> Copy for ServoLayoutDocument<'dom, LayoutDataType> {}
   |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0204]: the trait `std::marker::Copy` cannot be implemented for this type
  --> components/script/layout_dom/element.rs:68:54
   |
55 |     element: LayoutDom<'dom, Element>,
   |     --------------------------------- this field does not implement `std::marker::Copy`
...
68 | impl<'dom, LayoutDataType: LayoutDataTrait> Copy for ServoLayoutElement<'dom, LayoutDataType> {}
   |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0204]: the trait `std::marker::Copy` cannot be implemented for this type
  --> components/script/layout_dom/node.rs:73:54
   |
51 |     pub(super) node: LayoutDom<'dom, Node>,
   |     -------------------------------------- this field does not implement `std::marker::Copy`
...
73 | impl<'dom, LayoutDataType: LayoutDataTrait> Copy for ServoLayoutNode<'dom, LayoutDataType> {}
   |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0204]: the trait `std::marker::Copy` cannot be implemented for this type
  --> components/script/layout_dom/shadow_root.rs:30:54
   |
19 |     shadow_root: LayoutDom<'dom, ShadowRoot>,
   |     ---------------------------------------- this field does not implement `std::marker::Copy`
...
30 | impl<'dom, LayoutDataType: LayoutDataTrait> Copy for ServoShadowRoot<'dom, LayoutDataType> {}
   |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0204`.
error: could not compile `script` (lib) due to 4 previous errors

```